src/python/pants/backend/python/lint/pylint/rules.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from dataclasses import dataclass
+from typing import Optional, Tuple
+
+from pants.backend.python.lint.pylint.subsystem import Pylint
+from pants.backend.python.lint.python_lint_target import PythonLintTarget
+from pants.backend.python.rules.pex import (
+  CreatePex,
+  Pex,
+  PexInterpreterConstraints,
+  PexRequirements,
+)
+from pants.backend.python.rules.prepare_chrooted_python_sources import ChrootedPythonSources
+from pants.backend.python.subsystems.subprocess_environment import SubprocessEncodingEnvironment
+from pants.build_graph.address import Address
+from pants.engine.fs import Digest, DirectoriesToMerge, PathGlobs, Snapshot
+from pants.engine.isolated_process import ExecuteProcessRequest, FallibleExecuteProcessResult
+from pants.engine.legacy.graph import HydratedTarget, HydratedTargets
+from pants.engine.legacy.structs import PythonTargetAdaptor, TargetAdaptor
+from pants.engine.rules import UnionRule, rule, subsystem_rule
+from pants.engine.selectors import Get, MultiGet
+from pants.option.global_options import GlobMatchErrorBehavior
+from pants.python.python_setup import PythonSetup
+from pants.rules.core.lint import LintResult
+from pants.rules.core.strip_source_root import SourceRootStrippedSources
+
+
+@dataclass(frozen=True)
+class PylintTarget:
+  target: TargetAdaptor
+
+
+def generate_args(file_names: Tuple[str, ...], pylint: Pylint) -> Tuple[str, ...]:
+  args = []
+  if pylint.options.config is not None:
+    args.append(f"--config={pylint.options.config}")
+  args.extend(pylint.options.args)
+  args.extend(file_names)
+  return tuple(args)
+
+
+@rule(name="Lint using Pylint")
+async def lint(
+  wrapped_target: PylintTarget,
+  pylint: Pylint,
+  python_setup: PythonSetup,
+  subprocess_encoding_environment: SubprocessEncodingEnvironment,
+) -> LintResult:
+  if pylint.options.skip:
+    return LintResult.noop()
+
+  target = wrapped_target.target
+
+  hydrated_target = await Get[HydratedTarget](Address, target.address)
+  dependencies = await MultiGet(
+    Get[HydratedTarget](Address, dependency)
+    for dependency in hydrated_target.dependencies
+  )
```

Comment:
Consider adding a comment explaining that Pylint needs direct dependencies in the chroot to ensure that imports are valid. It doesn't actually lint those direct dependencies nor does it care about transitive dependencies.